### ISSUE TITLE
Arduino_Core_STM32 compatibility

### ISSUE THREAD

**Issue by AndKe** [Naguissa/uTimerLib#9]
I've used this library with STM32Duino library,   now I needed to switch to this, much better library:
https://github.com/stm32duino/Arduino_Core_STM32

- but uTimerLib is not compatible with it ?
(output from example)

```




/home/andre/Arduino/arduino-1.8.10/arduino-builder -dump-prefs -logger=machine -hardware /home/andre/Arduino/arduino-1.8.10/hardware -hardware /home/andre/.arduino15/packages -hardware /home/andre/Arduino/projects/hardware -tools /home/andre/Arduino/arduino-1.8.10/tools-builder -tools /home/andre/Arduino/arduino-1.8.10/hardware/tools/avr -tools /home/andre/.arduino15/packages -built-in-libraries /home/andre/Arduino/arduino-1.8.10/libraries -libraries /home/andre/Arduino/projects/libraries -fqbn=STM32:stm32:GenF1:pnum=BLUEPILL_F103C6,upload_method=swdMethod,xserial=generic,usb=none,xusb=FS,opt=osstd,rtlib=nano -ide-version=10810 -build-path /tmp/arduino -warnings=default -build-cache /tmp/arduino_cache_586316 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.CMSIS.path=/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1 -prefs=runtime.tools.CMSIS-5.5.1.path=/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1 -prefs=runtime.tools.STM32Tools.path=/home/andre/.arduino15/packages/STM32/tools/STM32Tools/1.3.1 -prefs=runtime.tools.STM32Tools-1.3.1.path=/home/andre/.arduino15/packages/STM32/tools/STM32Tools/1.3.1 -prefs=runtime.tools.arm-none-eabi-gcc.path=/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7 -prefs=runtime.tools.arm-none-eabi-gcc-8.2.1-1.7.path=/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7 -verbose /home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial/uTimerLib_setInterval_s_example_serial.ino
/home/andre/Arduino/arduino-1.8.10/arduino-builder -compile -logger=machine -hardware /home/andre/Arduino/arduino-1.8.10/hardware -hardware /home/andre/.arduino15/packages -hardware /home/andre/Arduino/projects/hardware -tools /home/andre/Arduino/arduino-1.8.10/tools-builder -tools /home/andre/Arduino/arduino-1.8.10/hardware/tools/avr -tools /home/andre/.arduino15/packages -built-in-libraries /home/andre/Arduino/arduino-1.8.10/libraries -libraries /home/andre/Arduino/projects/libraries -fqbn=STM32:stm32:GenF1:pnum=BLUEPILL_F103C6,upload_method=swdMethod,xserial=generic,usb=none,xusb=FS,opt=osstd,rtlib=nano -ide-version=10810 -build-path /tmp/arduino -warnings=default -build-cache /tmp/arduino_cache_586316 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.CMSIS.path=/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1 -prefs=runtime.tools.CMSIS-5.5.1.path=/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1 -prefs=runtime.tools.STM32Tools.path=/home/andre/.arduino15/packages/STM32/tools/STM32Tools/1.3.1 -prefs=runtime.tools.STM32Tools-1.3.1.path=/home/andre/.arduino15/packages/STM32/tools/STM32Tools/1.3.1 -prefs=runtime.tools.arm-none-eabi-gcc.path=/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7 -prefs=runtime.tools.arm-none-eabi-gcc-8.2.1-1.7.path=/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7 -verbose /home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial/uTimerLib_setInterval_s_example_serial.ino
Using board 'GenF1' from platform in folder: /home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0
Using core 'arduino' from platform in folder: /home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0
bash -c "[ -f /home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial/build_opt.h ] || (mkdir -p /tmp/arduino/sketch && touch /tmp/arduino/sketch/build_opt.h)"
Detecting libraries used...
/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7/bin/arm-none-eabi-g++ -mcpu=cortex-m3 -mthumb @/tmp/arduino/sketch/build_opt.h -c -Os -w -std=gnu++14 -ffunction-sections -fdata-sections -nostdlib -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -I/home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/avr -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32 -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/LL -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/hid -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/cdc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Inc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Src/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/STM32F1xx/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -w -x c++ -E -CC -DSTM32F1xx -DARDUINO=10810 -DARDUINO_BLUEPILL_F103C6 -DARDUINO_ARCH_STM32 "-DBOARD_NAME=\"BLUEPILL_F103C6\"" -DSTM32F103x6 -DHAL_UART_MODULE_ENABLED -I/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1/CMSIS/Core/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Source/Templates/gcc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/variants/PILL_F103XX /tmp/arduino/sketch/uTimerLib_setInterval_s_example_serial.ino.cpp -o /dev/null
Alternatives for uTimerLib.h: [uTimerLib@1.2.2]
ResolveLibrary(uTimerLib.h)
  -> candidates: [uTimerLib@1.2.2]
/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7/bin/arm-none-eabi-g++ -mcpu=cortex-m3 -mthumb @/tmp/arduino/sketch/build_opt.h -c -Os -w -std=gnu++14 -ffunction-sections -fdata-sections -nostdlib -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -I/home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/avr -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32 -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/LL -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/hid -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/cdc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Inc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Src/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/STM32F1xx/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -w -x c++ -E -CC -DSTM32F1xx -DARDUINO=10810 -DARDUINO_BLUEPILL_F103C6 -DARDUINO_ARCH_STM32 "-DBOARD_NAME=\"BLUEPILL_F103C6\"" -DSTM32F103x6 -DHAL_UART_MODULE_ENABLED -I/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1/CMSIS/Core/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Source/Templates/gcc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/variants/PILL_F103XX -I/home/andre/Arduino/projects/libraries/uTimerLib/src /tmp/arduino/sketch/uTimerLib_setInterval_s_example_serial.ino.cpp -o /dev/null
/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7/bin/arm-none-eabi-g++ -mcpu=cortex-m3 -mthumb @/tmp/arduino/sketch/build_opt.h -c -Os -w -std=gnu++14 -ffunction-sections -fdata-sections -nostdlib -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -I/home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/avr -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32 -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/LL -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/hid -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/cdc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Inc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Src/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/STM32F1xx/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -w -x c++ -E -CC -DSTM32F1xx -DARDUINO=10810 -DARDUINO_BLUEPILL_F103C6 -DARDUINO_ARCH_STM32 "-DBOARD_NAME=\"BLUEPILL_F103C6\"" -DSTM32F103x6 -DHAL_UART_MODULE_ENABLED -I/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1/CMSIS/Core/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Source/Templates/gcc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/variants/PILL_F103XX -I/home/andre/Arduino/projects/libraries/uTimerLib/src /home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp -o /dev/null
Generating function prototypes...
/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7/bin/arm-none-eabi-g++ -mcpu=cortex-m3 -mthumb @/tmp/arduino/sketch/build_opt.h -c -Os -w -std=gnu++14 -ffunction-sections -fdata-sections -nostdlib -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -I/home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/avr -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32 -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/LL -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/hid -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/cdc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Inc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Src/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/STM32F1xx/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -w -x c++ -E -CC -DSTM32F1xx -DARDUINO=10810 -DARDUINO_BLUEPILL_F103C6 -DARDUINO_ARCH_STM32 "-DBOARD_NAME=\"BLUEPILL_F103C6\"" -DSTM32F103x6 -DHAL_UART_MODULE_ENABLED -I/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1/CMSIS/Core/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Source/Templates/gcc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/variants/PILL_F103XX -I/home/andre/Arduino/projects/libraries/uTimerLib/src /tmp/arduino/sketch/uTimerLib_setInterval_s_example_serial.ino.cpp -o /tmp/arduino/preproc/ctags_target_for_gcc_minus_e.cpp
/home/andre/Arduino/arduino-1.8.10/tools-builder/ctags/5.8-arduino11/ctags -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives /tmp/arduino/preproc/ctags_target_for_gcc_minus_e.cpp
Compiling sketch...
/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7/bin/arm-none-eabi-g++ -mcpu=cortex-m3 -mthumb @/tmp/arduino/sketch/build_opt.h -c -Os -std=gnu++14 -ffunction-sections -fdata-sections -nostdlib -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -MMD -I/home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/avr -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32 -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/LL -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/hid -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/cdc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Inc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Src/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/STM32F1xx/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -DSTM32F1xx -DARDUINO=10810 -DARDUINO_BLUEPILL_F103C6 -DARDUINO_ARCH_STM32 "-DBOARD_NAME=\"BLUEPILL_F103C6\"" -DSTM32F103x6 -DHAL_UART_MODULE_ENABLED -I/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1/CMSIS/Core/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Source/Templates/gcc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/variants/PILL_F103XX -I/home/andre/Arduino/projects/libraries/uTimerLib/src /tmp/arduino/sketch/uTimerLib_setInterval_s_example_serial.ino.cpp -o /tmp/arduino/sketch/uTimerLib_setInterval_s_example_serial.ino.cpp.o
Compiling libraries...
Compiling library "uTimerLib"
/home/andre/.arduino15/packages/STM32/tools/arm-none-eabi-gcc/8.2.1-1.7/bin/arm-none-eabi-g++ -mcpu=cortex-m3 -mthumb @/tmp/arduino/sketch/build_opt.h -c -Os -std=gnu++14 -ffunction-sections -fdata-sections -nostdlib -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -MMD -I/home/andre/Arduino/projects/libraries/uTimerLib/examples/uTimerLib_setInterval_s_example_serial -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/avr -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32 -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/LL -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/hid -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino/stm32/usb/cdc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Inc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/STM32F1xx_HAL_Driver/Src/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/STM32F1xx/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src -DSTM32F1xx -DARDUINO=10810 -DARDUINO_BLUEPILL_F103C6 -DARDUINO_ARCH_STM32 "-DBOARD_NAME=\"BLUEPILL_F103C6\"" -DSTM32F103x6 -DHAL_UART_MODULE_ENABLED -I/home/andre/.arduino15/packages/STM32/tools/CMSIS/5.5.1/CMSIS/Core/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Include/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/system/Drivers/CMSIS/Device/ST/STM32F1xx/Source/Templates/gcc/ -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/cores/arduino -I/home/andre/.arduino15/packages/STM32/hardware/stm32/1.7.0/variants/PILL_F103XX -I/home/andre/Arduino/projects/libraries/uTimerLib/src /home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp -o /tmp/arduino/libraries/uTimerLib/uTimerLib.cpp.o
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp: In member function 'void uTimerLib::_attachInterrupt_us(long unsigned int)':
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:247:18: error: 'TIMER_CH1' was not declared in this scope
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                  ^~~~~~~~~
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:247:18: note: suggested alternative: 'TIMER_NUM'
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                  ^~~~~~~~~
                  TIMER_NUM
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:247:29: error: 'TIMER_OUTPUTCOMPARE' was not declared in this scope
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                             ^~~~~~~~~~~~~~~~~~~
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:247:29: note: suggested alternative: 'TIMER_OUTPUT_COMPARE'
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                             ^~~~~~~~~~~~~~~~~~~
                             TIMER_OUTPUT_COMPARE
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:249:35: error: 'class HardwareTimer' has no member named 'setPeriod'; did you mean 'setOverflow'?
   uint16_t timerOverflow = Timer3.setPeriod(us);
                                   ^~~~~~~~~
                                   setOverflow
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:250:10: error: 'class HardwareTimer' has no member named 'setCompare'; did you mean 'setCount'?
   Timer3.setCompare(TIMER_CH1, timerOverflow);
          ^~~~~~~~~~
          setCount
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp: In member function 'void uTimerLib::_attachInterrupt_s(long unsigned int)':
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:578:18: error: 'TIMER_CH1' was not declared in this scope
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                  ^~~~~~~~~
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:578:18: note: suggested alternative: 'TIMER_NUM'
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                  ^~~~~~~~~
                  TIMER_NUM
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:578:29: error: 'TIMER_OUTPUTCOMPARE' was not declared in this scope
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                             ^~~~~~~~~~~~~~~~~~~
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:578:29: note: suggested alternative: 'TIMER_OUTPUT_COMPARE'
   Timer3.setMode(TIMER_CH1, TIMER_OUTPUTCOMPARE);
                             ^~~~~~~~~~~~~~~~~~~
                             TIMER_OUTPUT_COMPARE
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:579:10: error: 'class HardwareTimer' has no member named 'setPeriod'; did you mean 'setOverflow'?
   Timer3.setPeriod((unsigned long int) 1000000);   // 1s, in microseconds
          ^~~~~~~~~
          setOverflow
/home/andre/Arduino/projects/libraries/uTimerLib/src/uTimerLib.cpp:580:10: error: 'class HardwareTimer' has no member named 'setCompare'; did you mean 'setCount'?
   Timer3.setCompare(TIMER_CH1, 1000000);
          ^~~~~~~~~~
          setCount
Multiple libraries were found for "uTimerLib.h"
 Used: /home/andre/Arduino/projects/libraries/uTimerLib
Using library uTimerLib at version 1.2.2 in folder: /home/andre/Arduino/projects/libraries/uTimerLib 
exit status 1
Error compiling for board Generic STM32F1 series.
```

**Comment by Naguissa** [comment]
I have no idea about that repository and why STM32duino is down...

I know STM32 / Arduino by STM32Duino.com (down) and Roger Clarckbourne, https://github.com/rogerclarkmelbourne/Arduino_STM32

I don't see him as contributor and your repository is not forked from Roger's one. In fact, "yours" is derived from ST development tools (Cube, etc) and "mine" is derived from Maple's Arduino-derived IDE.

I'll try your pointed repository and check if I can find its #defines in order to fix it....


If you want to try Roger's STM32 support, only to check, you can see the instructions I wrote here: https://translate.google.es/translate?sl=es&tl=no&u=https%3A%2F%2Fwww.foroelectro.net%2Felectronica-digital-microcontroladores-f8%2Fmicrocontroladores-basados-stm32-configuracion-arduino-ide-caracteristicas-t99.html


Cheers!

**Comment by AndKe** [comment]
Yes, I've used Roger's STM32duino till recently.
Now I find that Arduino_Core_STM32 to be much better, more "complete" and compatible.

I do not know what happened to STM32duino
What I do know, is that the STM32duino forum closed for new sign-up - the "register" button and page was removed.  Then after an update, the option was back, but the forum owner never authorized new registrations.  A month after that again, the forum, webpage has been removed completely.  I guess he just got tired or something like that.

In any case, the Arduino_Core_STM32 is at least very much alive and being worked on.

**Comment by Naguissa** [comment]
Oh, I had no idea....


I have a project where Maple Mini overclocked was getting too slow, I was thinking on start using asembler blocks.... So maybe with this new board support..... 


More arguments to try it ASAP.....


I'll check it, I think I'll need to check #define's and rename some functions; maybe check on old core files the constant definitions, but it shouldn't be too difficult.


Cheers!

**Comment by Naguissa** [comment]
Added compatibility with both STM32 cores.

New core has been tested on Maple Mini (and compiled for few other boards).

New release published, 1.3.0

It should be available on the IDE's library manager in one hour approximately.

**Comment by AndKe** [comment]
Thank you :)
